lte/gateway/c/li_agent/main.cpp

```diff
@@ -0,0 +1,117 @@
+
+#include "InterfaceMonitor.h"
+#include "PDUGenerator.h"
+#include "ProxyConnector.h"
+
+#define LIAGENTD "liagentd"
+#define LIAGENTD_VERSION "1.0"
+
+static magma::mconfig::LIAgentD get_default_mconfig() {
+  magma::mconfig::LIAgentD mconfig;
+  mconfig.set_log_level(magma::orc8r::LogLevel::INFO);
+  return mconfig;
+}
+
+static magma::mconfig::LIAgentD load_mconfig() {
+  magma::mconfig::LIAgentD mconfig;
+  magma::MConfigLoader loader;
+  if (!loader.load_service_mconfig(LIAGENTD, &mconfig)) {
+    MLOG(MERROR) << "Unable to load mconfig for liagentd, using default";
+    return get_default_mconfig();
+  }
+  return mconfig;
+}
+
+static uint32_t get_log_verbosity(
+    const YAML::Node& config, magma::mconfig::LIAgentD mconfig) {
+  if (!config["log_level"].IsDefined()) {
+    if (mconfig.log_level() < 0 || mconfig.log_level() > 4) {
+      return MINFO;
+    }
+    return mconfig.log_level();
+  }
+  std::string log_level = config["log_level"].as<std::string>();
+  if (log_level == "DEBUG") {
+    return MDEBUG;
+  } else if (log_level == "INFO") {
+    return MINFO;
+  } else if (log_level == "WARNING") {
+    return MWARNING;
+  } else if (log_level == "ERROR") {
+    return MERROR;
+  } else if (log_level == "FATAL") {
+    return MFATAL;
+  } else {
+    MLOG(MINFO) << "Invalid log level in config: "
+                << config["log_level"].as<std::string>();
+    return MINFO;
+  }
+}
+
+int main(void) {
+  magma::init_logging(LIAGENTD);
+
+  auto mconfig = load_mconfig();
+  auto config  = magma::ServiceConfigLoader{}.load_service_config(LIAGENTD);
+  magma::set_verbosity(get_log_verbosity(config, mconfig));
+  MLOG(MINFO) << "Starting LI Agent service";
+
+  // TODO: Ignore SigPipe as SSL_write throws that occasionally, investigate
+  sigset_t blockedSignal;
+  sigemptyset(&blockedSignal);
+  sigaddset(&blockedSignal, SIGPIPE);
+  pthread_sigmask(SIG_BLOCK, &blockedSignal, NULL);
+
+  auto directoryd_client = std::make_shared<magma::AsyncDirectorydClient>();
+  std::thread directoryd_response_handling_thread([&]() {
+    MLOG(MINFO) << "Started DirectoryD response thread";
+    directoryd_client->rpc_response_loop();
+  });
+
+  std::string interface_name = config["interface_name"].as<std::string>();
+  std::string pkt_dst_mac    = config["pkt_dst_mac"].as<std::string>();
+  std::string pkt_src_mac    = config["pkt_src_mac"].as<std::string>();
+  std::string proxy_addr     = config["proxy_addr"].as<std::string>();
+  int proxy_port             = config["proxy_port"].as<int>();
+  std::string cert_file      = config["cert_file"].as<std::string>();
+  std::string key_file       = config["key_file"].as<std::string>();
+
+  magma::service303::MagmaService server(LIAGENTD, LIAGENTD_VERSION);
+  server.Start();
+
+  auto proxy_connector = std::make_shared<magma::lte::ProxyConnector>(
+      proxy_addr, proxy_port, cert_file, key_file);
+  auto pkt_generator = std::make_shared<magma::lte::PDUGenerator>(
```

Comment:
I mentioned `auto` dis-use above in Google Style. To be clear - this use case is OK because the type is clearly encoded on the right hand side (same for make_unique). So this `auto` is totally fine.